dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java

```diff
@@ -148,10 +152,20 @@ public void removeIndicatorGroup( ProgramIndicatorGroup group )
      */
     public Boolean hasNonDefaultBoundaries()
     {
-        return this.analyticsPeriodBoundaries.size() != 2 || ( this.analyticsType == AnalyticsType.EVENT && 
-            !this.analyticsPeriodBoundaries.containsAll( defaultEventTypeBoundaries ) ||
+        return this.analyticsPeriodBoundaries.size() != 2 || ( this.analyticsType == AnalyticsType.EVENT &&
+            !this.analyticsPeriodBoundaries.containsAll( DEFAULT_EVENT_TYPE_BOUNDARIES ) ||
             this.analyticsType == AnalyticsType.ENROLLMENT );
     }
+
+    /**
+     * Checks if expression contains V{analytics_period_end} or V{analytics_period_start}. It will be use in conjunction with hasNonDefaultBoundaries() in order to
+     * split sql queries for each period provided.
+     * @return true if expression has analytics period variables.
+     */
+    public boolean hasAnalyticsVariables()
+    {
+        return ANALYTICS_VARIABLE_PATTERN.matcher( this.expression ).find();

```

Comment:
Great, but also include the filter, if the variable is used either in the expression or filter we need this handling.

Reply:
fixed